src/AppCache.php

```diff
@@ -6,34 +6,56 @@
  */
 namespace EzSystems\PlatformHttpCacheBundle;
 
-use FOS\HttpCacheBundle\SymfonyCache\EventDispatchingHttpCache;
 use EzSystems\PlatformHttpCacheBundle\Proxy\TagAwareStore;
-use EzSystems\PlatformHttpCacheBundle\Proxy\UserContextSubscriber;
+use EzSystems\PlatformHttpCacheBundle\Proxy\UserContextListener;
+use FOS\HttpCache\SymfonyCache\CacheInvalidation;
+use FOS\HttpCache\SymfonyCache\EventDispatchingHttpCache;
+use Symfony\Bundle\FrameworkBundle\HttpCache\HttpCache;
 use Symfony\Component\HttpKernel\HttpKernelInterface;
 use Symfony\Component\HttpFoundation\Request;
 use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpKernel\KernelInterface;
 
 /**
  * Custom AppCache.
  *
  * "deprecated" This and classes used here will be removed once this package moves to FosHttpCache 2.x.
  */
-class AppCache extends EventDispatchingHttpCache
+class AppCache extends HttpCache implements CacheInvalidation
 {
+    use EventDispatchingHttpCache {
+        handle as protected baseHandle;
+        invalidate as protected baseInvalidate;
+    }
+
+    public function __construct(KernelInterface $kernel, $cacheDir = null)
+    {
+        parent::__construct($kernel, $cacheDir);
+        $this->addSubscriber(new UserContextListener(['session_name_prefix' => 'eZSESSID']));
+    }
+
+    public function fetch(Request $request, $catch = false)
+    {
+        return parent::fetch($request, $catch);
+    }
+
     /**
      * {@inheritdoc}
      */
     protected function createStore()
     {
-        return new TagAwareStore($this->cacheDir ?: $this->kernel->getCacheDir() . '/http_cache');
+        return new TagAwareStore([
```

Comment:
can be dropped, foshttpcache 2.x brings it's own

Reply:
used `\Toflar\Psr6HttpCacheStore\Psr6Store` directly